fml/time/time_recorder.cc

```diff
@@ -0,0 +1,24 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/time/time_recorder.h"
+#include "flutter/fml/logging.h"
+
+namespace fml {
+
+TimeRecorder::TimeRecorder() : initialized_(false) {}
+
+void TimeRecorder::Update(fml::TimePoint time) {
+  std::scoped_lock lock(mutex_);
+  initialized_ = true;
+  time_ = time;
+}
+
+fml::TimePoint TimeRecorder::Get() {
+  std::scoped_lock lock(mutex_);
```

Comment:
Consider a reader/writer lock instead (there in one in `//flutter/fml/synchronization/shared_mutex.h`). We don't know what the contention is going to be for users of this utility moving forward. This is fine for now too though.